tests/benchmarks/test_tpch.py

```diff
@@ -0,0 +1,315 @@
+    lineitem_filtered["sum_qty"] = lineitem_filtered.l_quantity
+    lineitem_filtered["sum_base_price"] = lineitem_filtered.l_extendedprice
+    lineitem_filtered["avg_qty"] = lineitem_filtered.l_quantity
+    lineitem_filtered["avg_price"] = lineitem_filtered.l_extendedprice
+    lineitem_filtered["sum_disc_price"] = lineitem_filtered.l_extendedprice * (
+        1 - lineitem_filtered.l_discount
+    )
+    lineitem_filtered["sum_charge"] = (
+        lineitem_filtered.l_extendedprice
+        * (1 - lineitem_filtered.l_discount)
+        * (1 + lineitem_filtered.l_tax)
+    )
+    lineitem_filtered["avg_disc"] = lineitem_filtered.l_discount
+    lineitem_filtered["count_order"] = lineitem_filtered.l_orderkey
+    gb = lineitem_filtered.groupby(["l_returnflag", "l_linestatus"])
+
+    total = gb.agg(
+        {
+            "sum_qty": "sum",
+            "sum_base_price": "sum",
+            "sum_disc_price": "sum",
+            "sum_charge": "sum",
+            "avg_qty": "mean",
+            "avg_price": "mean",
+            "avg_disc": "mean",
+            "count_order": "count",
+        }
+    )
+
+    total.reset_index().compute().sort_values(["l_returnflag", "l_linestatus"])
+
+
+@pytest.mark.client("tpch")
+def test_query_2(client):
+    var1 = 15
+    var2 = "BRASS"
+    var3 = "EUROPE"
+
+    region_ds = read_data("region")
+    nation_filtered = read_data("nation")
+    supplier_filtered = read_data("supplier")
+    part_filtered = read_data("part")
+    partsupp_filtered = read_data("partsupp")
+
+    region_filtered = region_ds[(region_ds["r_name"] == var3)]
+    r_n_merged = nation_filtered.merge(
+        region_filtered, left_on="n_regionkey", right_on="r_regionkey", how="inner"
+    )
+    s_r_n_merged = r_n_merged.merge(
+        supplier_filtered,
+        left_on="n_nationkey",
+        right_on="s_nationkey",
+        how="inner",
+    )
+    ps_s_r_n_merged = s_r_n_merged.merge(
+        partsupp_filtered, left_on="s_suppkey", right_on="ps_suppkey", how="inner"
+    )
+    part_filtered = part_filtered[
+        (part_filtered["p_size"] == var1)
+        & (part_filtered["p_type"].astype(str).str.endswith(var2))
+    ]
+    merged_df = part_filtered.merge(
+        ps_s_r_n_merged, left_on="p_partkey", right_on="ps_partkey", how="inner"
+    )
+    min_values = merged_df.groupby("p_partkey")["ps_supplycost"].min().reset_index()
+    min_values.columns = ["P_PARTKEY_CPY", "MIN_SUPPLYCOST"]
+    merged_df = merged_df.merge(
+        min_values,
+        left_on=["p_partkey", "ps_supplycost"],
+        right_on=["P_PARTKEY_CPY", "MIN_SUPPLYCOST"],
+        how="inner",
+    )
+    merged_df[
+        [
+            "s_acctbal",
+            "s_name",
+            "n_name",
+            "p_partkey",
+            "p_mfgr",
+            "s_address",
+            "s_phone",
+            "s_comment",
+        ]
+    ].persist().sort_values("s_acctbal", ascending=False).head(100)
```

Comment:
The TPCH query 2 defines the sorting on multiple columns (see https://www.tpc.org/TPC_Documents_Current_Versions/pdf/TPC-H_v3.0.1.pdf)

![image](https://github.com/coiled/benchmarks/assets/8629629/0a4d0732-c052-43f5-85ba-f9ccf4524aac)